node/lib/util/push.js

```diff
@@ -93,24 +94,23 @@ exports.push = co.wrap(function *(repo, remoteName, source, target, force) {
     // First, push the submodules.
 
     let errorMessage = "";
-    const subRepos = yield SubmoduleUtil.getSubmoduleRepos(repo);
     const shas = yield SubmoduleUtil.getSubmoduleShasForBranch(repo, source);
-    const index = yield repo.index();
-    const urls = yield SubmoduleConfigUtil.getSubmodulesFromIndex(repo, index);
```

Comment:
Yes it is. I think as we've evolved our understanding, it's clear (to me) that the way I was doing this was wrong If I'm pushing commit XYZ, I think I should be using the URL configured at commit XYZ rather than something staged in the index. (Maybe I should have mentioned this in the commit message).